scripts/parse_manifest.py

```diff
@@ -26,112 +24,118 @@ def __init__(self, argv=None):
-
-        for udf in self.process.udf:
-            if udf.find(con_type) >= 0:
-                step_udfs_to_parse.add(udf)
+        for sample in self.samples:
+            if self.sample_column_udf == 'Plate Sample Name':
+                sample_dict[sample.name] = sample
+            if self.sample_column_udf == '2D Barcode':
+                sample_dict[sample.udf['2D Barcode']] = sample
 
         # create the list to contain the samples to be updated in a batch PUT to the API. This should match the number
         # of input artifacts
         samples_to_put = []
 
-        key_cell = key_column + str(current_row)
-        key_value = ws[key_cell].value
-
-        while key_value:
-            for udf in step_udfs_to_parse:
-                #remove the container type tag from the step UDF name so can be used to find the corresponding sample UDF
-                lims_udf = udf.replace(con_type, '')
-                #use the column indicated in the step UDF and the current row counter to identify which cell in
-                #the spreadsheet should be referenced
-                udf_cell = self.process.udf[udf] + str(current_row)
-                #check for udfs that need to be converted to strings as indicated by the [Str] tag in the step UDF name
-
-                if lims_udf.find('[Str]') >=0:
-                    lims_udf=lims_udf.replace('[Str]','')
-                    udf_value=str(ws[udf_cell].value)
-
+        sample_id = ws[self.sample_column + str(self.current_row)].value
+        while sample_id:
+            if sample_id not in sample_dict:
+                # Raise value error if sample present in manifest but not in LIMS
+                raise InvalidStepError(sample_id + " present in manifest but not present in LIMS.")
+
+            for udf in self.step_udfs_to_parse:
+                # remove the container type tag from the step UDF name so can be used to find the corresponding sample UDF
+                lims_udf = udf.replace(self.con_type, '')
+                # use the column indicated in the step UDF and the current row counter to identify which cell in
+                # the spreadsheet should be referenced
+                udf_cell = self.process.udf[udf] + str(self.current_row)
+                # check for udfs that need to be converted to strings as indicated by the [Str] tag in the step UDF name
+
+                if lims_udf.find('[Str]') >= 0:
+                    lims_udf = lims_udf.replace('[Str]', '')
+                    udf_value = str(ws[udf_cell].value)
                 else:
-                    udf_value=ws[udf_cell].value
+                    udf_value = ws[udf_cell].value
 
-                #parse the data from the spreadsheet into the sample dictionary
-                sample_dict[key_value].udf[lims_udf] = udf_value
+                if udf_value is None:
+                    self.warning('Value UDF %s is not set for sample %s (cell %s).', lims_udf, sample_id, udf_cell)
```

Comment:
What happens to the warning message?

Reply:
At the moment (and until we setup the logging properly) nothing.
Note that before it would have added `None` as the UDF value. Maybe we want this feature to stay as it was.

Reply:
Lets leave as is. Not adding none is preferable.